### PR TITLE
Issue #3129 - websocket module refactor

### PR DESCRIPTION
**Pull request by lachlan-roberts**
#3129
Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

### CODE REVIEW

**Comment by gregw** [comment]
@sbordet @joakime This PR is failing a few tests, but can you have a look at it now to see if you agree with the renaming.
@lachlan-roberts I'm assuming you wont get a chance to work on fixing the tests for 2 weeks?   So I may pull this into a branch and work on it myself as I relocate.

File: jetty-osgi/jetty-osgi-boot/jettyhome/etc/jetty.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
          <Item>org.eclipse.jetty.plus.webapp.EnvConfiguration</Item>
          <Item>org.eclipse.jetty.webapp.JmxConfiguration</Item>
          <Item>org.eclipse.jetty.websocket.server.JettyWebSocketConfiguration</Item>
-         <Item>org.eclipse.jetty.websocket.jsr356.JavaxWebSocketExtensionConfig</Item>
+         <Item>JavaxWebSocketExtensionConfig</Item>
```
Comment (joakime) [inline jetty-osgi/jetty-osgi-boot/jettyhome/etc/jetty.xml]: This is wrong.

`JavaxWebSocketExtensionConfig` is not a `Configuration` class.

File: jetty-osgi/test-jetty-osgi/src/test/config/etc/jetty-with-custom-class.xml [hunk 2]
```diff
@@ -82,7 +82,7 @@
          <Item>org.eclipse.jetty.plus.webapp.EnvConfiguration</Item>
          <Item>org.eclipse.jetty.webapp.JmxConfiguration</Item>
          <Item>org.eclipse.jetty.websocket.server.JettyWebSocketConfiguration</Item>
-         <Item>org.eclipse.jetty.websocket.jsr356.JavaxWebSocketExtensionConfig</Item>
+         <Item>org.eclipse.jetty.websocket.javax.common.JavaxWebSocketExtensionConfig</Item>
```
Comment (joakime) [inline jetty-osgi/test-jetty-osgi/src/test/config/etc/jetty-with-custom-class.xml]: Ditto.

`JavaxWebSocketExtensionConfig` is not a Jetty `Configuration` class.

Perhaps you want `org.eclipse.jetty.websocket.javax.server.JavaxWebSocketServerConfiguration` ?

**Comment by joakime** [comment]
a fair number of websocket osgi manifest fixes have been made in the `jetty-10.0.x-3162_jetty_servlet_api` branch.
should probably wait until that branch is reviewed / accepted / merged.

**Comment by gregw** [comment]
specially need to include f4cf671b27

**Comment by gregw** [comment]
I've merged this into branch https://github.com/eclipse/jetty.project/tree/jetty-10.0.x-3129-websocket to continue dev and testing.
I'll leave this PR open until the branch has addressed all the comments made.

**Pull request closed without merging**